source/MaterialXRenderGlsl/TextureBaker.h

```diff
@@ -243,6 +248,9 @@ class MX_RENDERGLSL_API TextureBaker : public GlslRenderer
     // Generate a texture filename for the given graph output.
     FilePath generateTextureFilename(const StringMap& fileTemplateMap);
 
+    // Create document that links shader outputs to a material.
+    DocumentPtr generateBakedDocumentFromShader(NodePtr shader, const StringVec& udimSet);
```

Comment:
Perhaps it would be clearer to name this method `generateNewDocumentFromShader`, since it doesn't actually perform any texture baking work.